radix-engine-common/src/macros.rs

```diff
@@ -29,32 +29,28 @@ macro_rules! dec {
     // NOTE: Decimal arithmetic operation safe unwrap.
     // In general, it is assumed that reasonable literals are provided.
     // If not then something is definitely wrong and panic is fine.
-    ($x:literal) => {
-        $crate::math::Decimal::try_from($x).unwrap()
-    };
+    ($x:literal) => {{
+        let (dec_string, dec_places) = radix_engine_macros::decimal_string!($x);
+        if dec_places > $crate::math::Decimal::SCALE {
+            panic!("Unsupported decimal places");
+        }
+        let multiplier: $crate::math::I192 =
+            $crate::math::I192::TEN.const_pow($crate::math::Decimal::SCALE - dec_places);
+        let b = $crate::math::I192::parse_str(dec_string);
+        $crate::math::Decimal(b.const_mul(multiplier))
```

Comment:
Why not moving all the computation to the macro?

Reply:
Done